src/components/OptionsList/BaseOptionsList.js

```diff
@@ -36,281 +36,283 @@ const defaultProps = {
-
-    /**
-     * This helper function is used to memoize the computation needed for getItemLayout. It is run whenever section data changes.
-     *
-     * @returns {Array<Object>}
-     */
-    const buildFlatSectionArray = () => {
-        let offset = 0;
-
-        // Start with just an empty list header
-        const flatArray = [{length: 0, offset}];
-
-        // Build the flat array
-        for (let sectionIndex = 0; sectionIndex < sections.length; sectionIndex++) {
-            const section = sections[sectionIndex];
-            // Add the section header
-            const sectionHeaderHeight = section.title && !hideSectionHeaders ? variables.optionsListSectionHeaderHeight : 0;
-            flatArray.push({length: sectionHeaderHeight, offset});
-            offset += sectionHeaderHeight;
-
-            // Add section items
-            for (let i = 0; i < section.data.length; i++) {
-                let fullOptionHeight = variables.optionRowHeight;
-                if (i > 0 && shouldHaveOptionSeparator) {
-                    fullOptionHeight += variables.borderTopWidth;
+const viewabilityConfig = {viewAreaCoveragePercentThreshold: 95};
+
+const BaseOptionsList = forwardRef(
+    (
+        {
+            keyboardDismissMode,
+            onScrollBeginDrag,
+            onScroll,
+            listStyles,
+            focusedIndex,
+            selectedOptions,
+            headerMessage,
+            isLoading,
+            sections,
+            onLayout,
+            hideSectionHeaders,
+            shouldHaveOptionSeparator,
+            showTitleTooltip,
+            optionHoveredStyle,
+            sectionHeaderStyle,
+            showScrollIndicator,
+            contentContainerStyles: contentContainerStylesProp,
+            listContainerStyles: listContainerStylesProp,
+            shouldDisableRowInnerPadding,
+            shouldPreventDefaultFocusOnSelectRow,
+            disableFocusOptions,
+            canSelectMultipleOptions,
+            shouldShowMultipleOptionSelectorAsButton,
+            multipleOptionSelectorButtonText,
+            onAddToSelection,
+            highlightSelectedOptions,
+            onSelectRow,
+            boldStyle,
+            isDisabled,
+            isRowMultilineSupported,
+            isLoadingNewOptions,
+            nestedScrollEnabled,
+            bounces,
+            renderFooterContent,
+            safeAreaPaddingBottomStyle,
+        },
+        innerRef,
+    ) => {
+        const styles = useThemeStyles();
+        const flattenedData = useRef();
+        const previousSections = usePrevious(sections);
+        const didLayout = useRef(false);
+
+        const listContainerStyles = useMemo(() => listContainerStylesProp || [styles.flex1], [listContainerStylesProp, styles.flex1]);
+        const contentContainerStyles = useMemo(() => [contentContainerStylesProp, safeAreaPaddingBottomStyle], [contentContainerStylesProp, safeAreaPaddingBottomStyle]);
```

Comment:
```suggestion
        const contentContainerStyles = useMemo(() => [safeAreaPaddingBottomStyle, ...contentContainerStylesProp], [contentContainerStylesProp, safeAreaPaddingBottomStyle]);
```

Btw, any reason for moving `safeAreaPaddingBottomStyle` from parent component to here?

Reply:
The reason to move `safeAreaPaddingBottomStyle ` to child is to avoid the unnecessary re-render due to new array instance being created on each render. Below is how it was done previously in the parent component.

`contentContainerStyles={[safeAreaPaddingBottomStyle, ...this.props.contentContainerStyles]}`

Now we construct the `contentContainerStyles` in child component.

Reply:
> Your current change will cause regression if contentContainerStylesProp includes paddingBottom.
i.e. TagPicker, ReportParticipantsPage

With your suggested change we can avoid this, right? I also see in my changes that I missed to spread the object, which you have highlighted.

Reply:
Just spreading is fine I think